modules/phase_field/src/mesh/MortarPeriodicMesh.C

```diff
@@ -89,50 +46,36 @@ MortarPeriodicMesh::clone() const
 void
 MortarPeriodicMesh::buildMesh()
 {
-  // hard code teh elem type for now (otherwise I have to duplicate too much stuff from libmesh)
-  ElemType elem_type_2D = Utility::string_to_enum<ElemType>("QUAD4");
-
-  switch (_dim)
-  {
-    case 2:
-      // build the main mesh
-      MeshTools::Generation::build_square(dynamic_cast<UnstructuredMesh &>(getMesh()),
-                                          _nx, _ny,
-                                          _xmin, _xmax,
-                                          _ymin, _ymax,
-                                          elem_type_2D);
-
-      // build the right mortar interface
-      addLineMesh(_ny, _xmax * 1.01, _ymin, _xmax * 1.01, _ymax, _block_x);
-
-      // build the top mortar interface
-      addLineMesh(_nx, _xmin, _ymax * 1.01, _xmax, _ymax * 1.01, _block_y);
-      break;
-
-    case 3: // not yet implemented
-    default:
-      mooseError("Only 2D meshes are supported.");
-  }
-  _us_mesh.prepare_for_use(/*skip_renumber =*/ false);
-}
-
-void
-MortarPeriodicMesh::addLineMesh(unsigned int nelem, Real x0, Real y0, Real x1, Real y1, subdomain_id_type id)
-{
-  dof_id_type max_node_id = _us_mesh.max_node_id() + 1;
-
-  // add nodes
-  for (unsigned int i = 0; i <= nelem; ++i)
-    _us_mesh.add_point (Point(static_cast<Real>(i) / nelem * (x1 - x0) + x0,
-                              static_cast<Real>(i) / nelem * (y1 - y0) + y0,
-                              0), max_node_id + i);
-
-  // add edges
-  for (unsigned int i = 0; i < nelem; ++i)
-  {
-    Elem * elem = _us_mesh.add_elem(new Edge2);
-    elem->subdomain_id() = id;
-    elem->set_node(0) = _us_mesh.node_ptr(max_node_id + i);
-    elem->set_node(1) = _us_mesh.node_ptr(max_node_id + i + 1);
-  }
+  // build the main mesh
+  GeneratedMesh::buildMesh();
+
+  // boundaries
+  const std::vector<BoundaryName> boundary_names = { "left", "bottom", "back" };
+
+  buildBndElemList();
+
+  // build side meshes
+  for (unsigned short i = 0; i < _dim; ++i)
+    if (_periodic_dirs.contains(i))
+    {
+      BoundaryID current_boundary_id = getBoundaryID(boundary_names[i]);
+
+      for (auto it = bndElemsBegin(); it != bndElemsEnd(); ++it)
```

Comment:
Since you're using C++11 here anyway... this should probably be a range-based-for